src/DynamicRupture/FrictionLaws/FrictionSolverCommon.h

```diff
@@ -235,6 +275,41 @@ inline void postcomputeImposedStateFromNewStress(
   }
 }
 
+/**
+ * adjusts initial stresses based on the given nucleation ones
+ *
+ * @param[out] initialStressInFaultCS
+ * @param[in] nucleationStressInFaultCS
+ * @param[in] t0
+ * @param[in] dt
+ * @param[in] index - device iteration index
+ */
+template <RangeType Type = RangeType::CPU,
+          typename MathFunctions = seissol::functions::HostStdFunctions>
+inline void adjustInitialStress(real initialStressInFaultCS[misc::numPaddedPoints][6],
+                                const real nucleationStressInFaultCS[misc::numPaddedPoints][6],
+                                real fullUpdateTime,
+                                real t0,
+                                real dt,
+                                unsigned startIndex = 0) {
+  if (fullUpdateTime <= t0) {
+    const real gNuc =
+        gaussianNucleationFunction::smoothStepIncrement<MathFunctions>(fullUpdateTime, dt, t0);
+
+    using Range = typename NumPoints<Type>::Range;
+
+#ifndef ACL_DEVICE
+    #pragma omp simd
+#endif
+    for (auto index = Range::start; index < Range::end; index += Range::step) {
+      auto pointIndex{startIndex + index};
```

Comment:
Also consider to put this into the loop header.

Reply:
discussed